### ISSUE TITLE
Single Row Action

### ISSUE THREAD

**Issue by gondomar** [vmware/clarity#2435]
**Select one ...** (check one with "x")

```
[ ] bug
[X] feature request
[ ] enhancement
```

### Expected behavior

I can't get it to work.  The documentation appears to be incomplete.

### Actual behavior

I can't get it to work.

### Reproduction of behavior

* Include a link to the reproduction scenario you created by forking one of the Clarity Plunker Templates:
  
  
  
  

### Environment details

* **Angular version:** 4.x.x

* **Clarity version:**

* **OS and version:**

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

**Comment by mathisscott** [comment]
Hi @gondomar 
Can you try replicating your issue in a [stackblitz](https://stackblitz.com/edit/clarity-light-theme-v012-bsnf7z?file=src/main.ts)?

We are often happy to help people work through issues. But we don't know where to start with this one.

**Comment by gondomar** [comment]
What I'm trying to do is quite straight-forward, but...
https://clarity-light-theme-v012-wjhsmo.stackblitz.io

**Comment by gnomeontherun** [comment]
If you want to select an item, you should use the selection options of the datagrid. Otherwise, you haven't implemented any logic to handle the click events for the edit/delete buttons, which is on the application side to manage. 

https://vmware.github.io/clarity/documentation/v0.12/datagrid/batch-action

**Comment by gondomar** [comment]
I don't want to use the selection.  You'll notice that the action-overflow items have "edit" and "delete".  I simply want those records to populate a form on the same screen.  That means that it needs to have access to that particular item.  The example on clarity found here:
https://vmware.github.io/clarity/documentation/v0.12/datagrid/single-action
does exactly that, but the documentation doesn't match what it's doing - or simply isn't complete.

**Comment by gnomeontherun** [comment]
This is really a question for Stack Overflow as you are the one who has to populate the list of records when someone clicks the edit/delete buttons. You define those buttons and their behaviors, not Clarity. So unless you use our built in selection, Clarity does not try to handle selecting items in some way. Our documentation doesn't show the click handler, because thats up to the application to handle. You can view the source of the demo here https://github.com/vmware/clarity/blob/website/latest/src/app/documentation/demos/datagrid/single-action/single-action.ts.

So basically, your (click) handlers on the button should pass the row object and manage the logic to display a form based on that row. 

I'm closing this as a question and not as a Clarity issue.
